Documentation/network/lb-ipam.rst

```diff
@@ -0,0 +1,382 @@
+conditions in the status.
+
+The service conditions can be checked like so:
+
+.. code-block:: shell-session
+
+    kubectl -n example get svc/service-red -o jsonpath='{.status.conditions}' | jq
+    [
+      {
+        "lastTransitionTime": "2022-10-06T13:40:48Z",
+        "message": "There are no enabled CiliumLoadBalancerIPPools that match this service",
+        "reason": "no_pool",
+        "status": "False",
+        "type": "io.cilium/lb-ipam-request-satisfied"
+      }
+    ]
+
+After updating the service labels to match our ``blue-pool`` from before we see:
+
+.. code-block:: shell-session
+
+    $ kubectl -n example get svc
+    NAME          TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
+    service-red   LoadBalancer   10.96.192.212   20.0.10.163   1234:30628/TCP   12m
+
+    $ kubectl -n example get svc/service-red -o jsonpath='{.status.conditions}' | jq
+    [
+      {
+        "lastTransitionTime": "2022-10-06T13:40:48Z",
+        "message": "There are no enabled CiliumLoadBalancerIPPools that match this service",
+        "reason": "no_pool",
+        "status": "False",
+        "type": "io.cilium/lb-ipam-request-satisfied"
+      },
+      {
+        "lastTransitionTime": "2022-10-06T13:52:55Z",
+        "message": "",
+        "reason": "satisfied",
+        "status": "True",
+        "type": "io.cilium/lb-ipam-request-satisfied"
+      }
+    ]
+
+IPv4 / IPv6 families + policy
+-----------------------------
+
+LB IPAM supports IPv4 and/or IPv6 in SingleStack or `DualStack <https://kubernetes.io/docs/concepts/services-networking/dual-stack/>`__ mode. 
+Services can use the ``.spec.ipFamilyPolicy`` and ``.spec.ipFamilies`` fields to change
+the requested IPs.
+
+If ``.spec.ipFamilyPolicy`` isn't specified, ``SingleStack`` mode is assumed. 
+If both IPv4 and IPv6 are enabled in ``SingleStack`` mode, an IPv4 address is allocated.
+
+If ``.spec.ipFamilyPolicy`` is set to ``PreferDualStack``, LB IPAM will attempt to allocate 
+both an IPv4 and IPv6 address if both are enabled on the cluster. If only IPv4 or only IPv6 is
+enabled on the cluster, the service is still considered "satisfied".
+
+If ``.spec.ipFamilyPolicy`` is set to ``RequireDualStack`` LB IPAM will attempt to allocate
+both an IPv4 and IPv6 address. The service is considered "unsatisfied" If IPv4 
+or IPv6 is disabled on the cluster.
+
+The order of ``.spec.ipFamilies`` has no effect on LB IPAM but is significant for cluster IP
+allocation which isn't handled by LB IPAM.
+
+LoadBalancerClass
+-----------------
+
+Kubernetes >= v1.24 supports `multiple load balancers <https://kubernetes.io/docs/concepts/services-networking/service/#load-balancer-class>`_ 
+in the same cluster. Picking between load balancers is done with the ``.spec.loadBalancerClass`` field. 
+When LB IPAM is enabled it assumed that it is supposed to allocate and assign IPs for services with 
+no load balancer class set.
```

Comment:
Should we be able to tie pools with classes?

Reply:
Perhaps some day. We currently only have 1 LB class which is for the upcoming service announcement feature of the BGP Control plane. We can always add this later when we add additional loadbalancers which utilize LB-IPAM